### DISCUSSION TITLE
which ip address to assign to xfrm interfaces

### DISCUSSION THREAD

**Discussion by nirdothan** [strongswan/strongswan#1940] [Q&A] [answered]
Hi,
In route based VPN documentation it doesn't seem that after creation of xfrm interfaces and IP address needs to be assigned to them, however without an ip address it seems that packets are not routed to the xfrm interfaces. It does seem to work when I assign $PLUTO_MY_CLIENT. What is the guidance in that respect?

**Comment by tobiasbrunner** [comment, marked as answer]
No IP is necessary on the XFRM interface. If packets sent from the host itself need to have a specific source IP to match the IPsec policies, you can set a preferred source address in the route (`src` in `ip route`) so it doesn't matter on which interface that IP is located.
